helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java

```diff
@@ -765,12 +765,17 @@ public Response getClusterMaintenanceHistory(@PathParam("clusterId") String clus
   @GET
   @Path("{clusterId}/controller/maintenanceSignal")
   public Response getClusterMaintenanceSignal(@PathParam("clusterId") String clusterId) {
-    HelixDataAccessor dataAccessor = getDataAccssor(clusterId);
-    MaintenanceSignal maintenanceSignal =
-        dataAccessor.getProperty(dataAccessor.keyBuilder().maintenance());
-    if (maintenanceSignal != null) {
-      Map<String, String> maintenanceInfo = maintenanceSignal.getRecord().getSimpleFields();
+    boolean inMaintenanceMode = getHelixAdmin().isInMaintenanceMode(clusterId);
+
+    if (inMaintenanceMode) {
+      HelixDataAccessor dataAccessor = getDataAccssor(clusterId);
+      PropertyKey maintenanceKey = dataAccessor.keyBuilder().maintenance();
+      MaintenanceSignal maintenanceSignal = dataAccessor.getProperty(maintenanceKey);
```

Comment:
NIT: combine these two line since maintenanceKey is not used for following code. No need for defining a new variable.

Reply:
Good point, thanks. I've addressed this in newest commit